test/unit/test_vfs2.c

```diff
@@ -339,6 +362,9 @@ TEST(vfs2, startup_both_nonempty, set_up, tear_down, 0, NULL)
 	return MUNIT_OK;
 }
 
+/**
+ * Single-node test of rolling back a transaction.
+ */
 TEST(vfs2, rollback, set_up, tear_down, 0, NULL)
```

Comment:
Yes, it's too weak. Right now all it accomplishes is covering the codepath that frees any pending transaction when the write lock is released without `COMMIT_PHASETWO`, but we should also assert that the transaction isn't visible at the end. And that reminds me, this test should also use a much larger transaction, because otherwise SQLite won't actually write any WAL frames until commit time...